### DISCUSSION TITLE
Syncing a Fork Repo.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21892] [New to GitHub]
<p>To sync my fork with the main repo someone mentioned:</p>
<p>Click “New Pull Request”, then set the “head” branch and fork you want to pull the changes from, and set the “base” branch and fork you want to pull the changes <em>to.</em></p>
<p> Click “Compare across forks” to do so</p>
<p>I am trying to understand is this process done on the actual github master repo or on the actual github fork repo (or either?)</p>
<p>Thank you!<br>
Joshei</p>

**Comment by eddiejaoude** [comment]
<p>To contribute back to the original repo that you forked from, you need to create the pull request on the original repo. </p>
<p>Here are the official docs from GitHub on Pull Requests <a href="https://help.github.com/articles/about-pull-requests/" rel="nofollow noopener">https://help.github.com/articles/about-pull-requests/ </a></p>
<p>I hope that helps</p>

**Comment by ghost** [comment]
<p>Yout link is broken.  As I am understanging there are two types of Pull Requests.  There is the one that you mentioned to contribute back and there is also a pull request to update the fork from the main repo.  It is the second Pull Request I am interested in when I use Compare Across Forks.  I am wondering which site I should choose the head and base on so that I have a proper sync (the main or the fork.)</p>
<p>Thank you,</p>
<p>Joshei</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/joshei">@joshei</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the How to use Git and GitHub board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Cheers!</p>

**Comment by strafe** [comment, marked as answer]
<p><a class="mention" href="/joshei">@joshei</a> Let me know if I’ve misinterpreted your question, as I’ve read it you want to sync your fork of a remote repository through GitHub’s UI with a pull request? If so I’ll list the steps to do so below:</p>
<p>As an example let’s say I’ve previously forked GitHub’s <a href="https://github.com/github/gemoji" rel="nofollow noopener">gemoji</a> repository and now my fork is outdated, I’d see something like this when navigating to my own fork: </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/96039ef280e520972a79f81dfb6f145df8271aaa.png" data-download-href="/uploads/short-url/lp5ns7Ben3ZGqtZmDQMmdG0NiNc.png?dl=1" title="KmZCs1y.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/15663890/181067451-e449ddbe-3368-41e1-8e13-8f37aff7a5d7.png" alt="KmZCs1y.png" data-base62-sha1="lp5ns7Ben3ZGqtZmDQMmdG0NiNc" width="690" height="26" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/96039ef280e520972a79f81dfb6f145df8271aaa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">KmZCs1y.png</span><span class="informations">977×37 4.07 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>  </p>
<p>From here you want to click <strong>Compare</strong> , GitHub will tell you _“There isn’t anthing to compare.” _so click on the hyperlink text saying <strong>Try switching the base for your comparison.</strong>. Now you’ve inverted base and head of the comparison you can click <strong>Create pull request</strong> , give it a name, create the request and merge it in.</p>

**Comment by ghost** [comment]
<p>That’s what I was wondering, the site is my own fork that I use!</p>
<p>Thanks,</p>
<p>Joshei</p>
